src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -466,6 +466,54 @@ def log_model_sparsification(self):
             f"all sparsification info: {sparsification_info}"
         )
 
+    def save_sample_inputs_outputs(
+        self, num_samples_to_export: int = 100, output_dir: Optional[str] = None
+    ):
+        """
+        Save sample inputs/outputs/labels in save_dir as .npz arrays
+
+        :param num_samples_to_export: Number of samples to export.
+            Defaults to 100
+        :param output_dir: The directory to store sample inputs and outputs in
+        """
+        num_samples = 0
+
+        potential_output_dir = (
+            self.args.output_dir if hasattr(self.args, "output_dir") else ""
+        )
+        output_dir = output_dir or potential_output_dir or ""
+
+        sample_inputs = os.path.join(output_dir, "sample-inputs")
+        sample_outputs = os.path.join(output_dir, "sample-outputs")
+        os.makedirs(sample_inputs, exist_ok=True)
+        os.makedirs(sample_outputs, exist_ok=True)
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        dataloader = self.get_train_dataloader()
+        _LOGGER.info(f"Exporting {num_samples_to_export} samples to {output_dir}")
+        for _, sample_batch in enumerate(dataloader):
+            sample_batch.pop("labels", None)
+            input_names = list(sample_batch.keys())
```

Comment:
Should filter names by the model forward functions arg spec (see tuan's implementation)

Reply:
It uses the same input names as tuan's function

[On this line](https://github.com/neuralmagic/transformers/blob/644c8faf4228f093d868dc6496a7081a450e98ba/src/transformers/sparse.py#L216)

Reply:
sounds good. let's keep an eye for incompatibilities with exported model